autotest/gcore/tiff_write.py

```diff
@@ -7331,6 +7331,42 @@ def test_tiff_write_jpeg_incompatible_of_paletted():
 
 
 ###############################################################################
+# Test blocksize overriding while creating (internal) overviews
+# on a newly created dataset
+
+def test_tiff_write_186_blocksize():
+    md = gdaltest.tiff_drv.GetMetadata()
+    if md['DMD_CREATIONOPTIONLIST'].find('WEBP') == -1:
+        pytest.skip()
+
+    blockSizes = [128,256]
+    for blockSize in blockSizes:
+        src_ds = gdal.Open('../gdrivers/data/utm.tif')
+        fname = 'tmp/tiff_write_186_bs%d.tif' % blockSize
+
+        ds = gdal.GetDriverByName('GTiff').Create(fname, 1024, 1024, 3,
+                                                  options=['COMPRESS=WEBP', 'TILED=YES',
+                                                      'BLOCKXSIZE=512', 'BLOCKYSIZE=512'])
+
+        data = src_ds.GetRasterBand(1).ReadRaster(0, 0, 512, 512, 1024, 1024)
+        ds.GetRasterBand(1).WriteRaster(0, 0, 1024, 1024, data)
+        ds.GetRasterBand(2).WriteRaster(0, 0, 1024, 1024, data)
+        ds.GetRasterBand(3).WriteRaster(0, 0, 1024, 1024, data)
+        with gdaltest.config_options({
+            'GDAL_TIFF_OVR_BLOCKSIZE':'%d'%blockSize,
+            'WEBP_LEVEL_OVERVIEW':'30',
```

Comment:
WEBP_LEVEL_OVERVIEW and GDAL_NUM_THREADS unneeded if opting for uncompressed case

Reply:
I've left a GDAL_NUM_THREADS testcase as the codepaths differ in that case

Reply:
except that threading is not use for the uncompressed path since there's no gain but loss to expect. Perhaps use LZW compression then which will always be built in

Reply:
doh! fixed